lib/AST/DeclContext.cpp

```diff
@@ -263,6 +263,25 @@ DeclContext *DeclContext::getInnermostSkippedFunctionContext() {
   return nullptr;
 }
 
+DeclContext *DeclContext::getInnermostClosureForSelfCapture() {
```

Comment:
Updated -- when I first added this I tried to import `Expr.h`, which didn't work because it was a circular reference. Just noticed the `namespace swift {` that we can use for a forward declaration.